server/middleware/userMiddlewares.js

```diff
@@ -9,14 +12,38 @@ import {
 } from "../utils/validators";
 import { status } from "../constants/middlewareConstants";
 import { passwordError } from "../constants/controllerConstants";
+
 const { FAIL } = status;
 import resp from "../constants/middlewareConstants";
 
 // checks whether the user is registered already
 export const checkEmail = async (req, res, next) => {
   const user = await User.model.findOne({ email: req.body.email });
   if (!user) {
-    next();
+    const groupIds = req.body.groups;
+    if(groupIds.length===0){
+      next();
+    }else{
+      for (let i = 0; i < groupIds.length; i++) {
+        if (!mongoose.Types.ObjectId.isValid(groupIds[i])) {
+          return res.status(422).json({
+            status: FAIL,
+            message: "The request is invalid."
```

Comment:
Kindly help make this message descriptive, it's hard to know what went wrong. 
You could also have a file where you store those messages as constants, so they can be changed easily.
I think this is meant to fail when one of the group id supplied in the `groups` field is invalid
If so, something in the line of the suggestion below should be fine
```suggestion
            message: "Groups contain invalid group id(s)"
```

Reply:
Or better for the sake of the user of the user - abstraction - `The groups do not exist on the system`.

Reply:
Btw this should be an impossible occurence unless someone is trying something fishy - you are selecting the groups to add from a list. Not keying in Ids.

Reply:
I will work on this as fix the changes that @shemogumbe requested